src/main.cpp

```diff
@@ -1,76 +1,68 @@
 #include <Arduino.h>
 #include <dataqueue.h>
-#include <display.h>
+#include <led.h>
 #include "ap.h"
 #include "gateway.h"
 #include "loramesh.h"
+#include "monitor.h"
+#include "wifinode.h"
 
-#define BLUE_LED 2
-
-#define OLED_SDA 4
-#define OLED_SCL 15
-#define OLED_RST 16
-#define OLED_SCREEN_WIDTH 128
-#define OLED_SCREEN_HEIGHT 64
+// Observability defines
+#ifndef MONITOR_CLASS
+#define MONITOR_CLASS LogMonitor
+#endif
+#ifndef MONITOR_UPDATE_INTERVAL
+#define MONITOR_UPDATE_INTERVAL 1000
+#endif
+#ifndef LED_TO_BLINK
+#define LED_TO_BLINK -1
+#endif
 
+// Operational defines
+#ifndef DATA_QUEUE_LENGTH
 #define DATA_QUEUE_LENGTH 10
+#endif
+#ifndef ACCESS_POINT_SSID
 #define ACCESS_POINT_SSID "MeshConnect"
+#endif
+#ifndef GATEWAY_SSID
 #define GATEWAY_SSID "MeshConnectGWAP"
+#endif
+#ifndef GATEWAY_SSID_PASSWORD
+#define GATEWAY_SSID_PASSWORD nullptr
+#endif
 
-Display* display;
 DataQueue<message_t>* wifiToLoraQueue;
 DataQueue<message_t>* loraToWifiQueue;
 WifiNode* wifi;
 LoraMesh* mesh;
+Monitor* monitor;
+Led led(LED_TO_BLINK);
 
 void setup() {
   Serial.begin(115200);
-  pinMode(BLUE_LED, OUTPUT);
-  digitalWrite(BLUE_LED, HIGH);
-
-  display = new Display(OLED_SDA, OLED_SCL, OLED_RST, OLED_SCREEN_WIDTH,
-                        OLED_SCREEN_HEIGHT);
+  led.on();
```

Comment:
Here you are using the LED as a normal one, its not blinking, its just turning on and off normally. I think it should be better for the define to be LED_ENABLED

Reply:
We use this LED to "notify" the user when the main loop function is being executed. In practice, the LED blinks every second :)

![image](https://user-images.githubusercontent.com/1025234/113596564-e1fc8d00-9610-11eb-87e8-d3111e40dc1d.png)

Reply:
Ok, its just that if someone uses LED_BLINK and don't set it up to turn on and off, it won't blink.